rust/lance/src/io/commit/dynamodb.rs

```diff
@@ -91,6 +101,23 @@ impl DynamoDBExternalManifestStore {
         table_name: &str,
         commiter_name: &str,
     ) -> Result<Arc<dyn ExternalManifestStore>> {
+        lazy_static::lazy_static! {
+            static ref SANITY_CHECK_CACHE: RwLock<HashSet<String>> = RwLock::new(HashSet::new());
```

Comment:
I could change this to a Lru Cache, this is is probably fine. It's very unlikly someone uses thousands of different dynamodb table for commit in the same process